### PR TITLE
docs: Add API Clients landing page

### PR DESCRIPTION
**Pull request by HonzaTuron**
Closes apify/apify-web#3622

Adds landing page for API clients (Python and JS). Menu item not added yet

**Differences design-wise:**
- Wasn't able to figure out background svg in hero section 
- In that case, not sure if that transparent section border makes sense

### CODE REVIEW

File: src/pages/api/index.tsx [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+            InternalLink: (props) => <Link {...props} />,
+            windowLocationHost: useBaseUrl(''),
+            isHrefTrusted: () => true,
+        }}>
+            <Layout>
+                <HeroWrapper
+                    heading="Apify API"
+                    isCentered
+                    description={<>Apify API provides useful features like automatic retries and convenience functions.<br/>
+                It connects you with any Actors and their tasks either in the Store or your own.</>}
```
Comment (jbartadev) [inline src/pages/api/index.tsx:114]: The copy doesn't make sense here, I know it's in the design but we need to change it to something connected to API in general.

I'd probably do something short and simple like:

Apify API provides programmatic access to the [Apify platform](https://docs.apify.com/).

What do you think?

File: src/pages/api/index.tsx [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+                            </TabTitleWrapper>,
+                            content: (
+                                <SectionWrapper
+                                    heading="JavaScript API client"
+                                    description={<div className="Description">
+                                    The official library to interact with Apify API from a web browser, Node.js, JavaScript, or Typescript applications.
+                                        <GitHubButton href="https://github.com/apify/apify-client-js" data-color-scheme="no-preference: light; light: light; dark: dark;" data-size="large" data-show-count="true">Star</GitHubButton>
+                                        <div className="DescriptionLinks">
+                                            <Button to='https://docs.apify.com/api/client/js/docs'>Get started</Button>
+                                            <ActionLink to='https://docs.apify.com/api/client/js'>JavaScript client reference</ActionLink>
```
Comment (jbartadev) [inline src/pages/api/index.tsx:173]: Let's remove the external icons from those

Snímek obrazovky 2024-05-05 v 12 26 55

File: src/pages/api/index.tsx [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+                                    sources={{ dark: useBaseUrl('/img/javascript-40x40.svg'), light: useBaseUrl('/img/javascript-40x40.svg') }}
+                                />
+                                JavaScript Client
+                            </TabTitleWrapper>,
+                            content: (
+                                <SectionWrapper
+                                    heading="JavaScript API client"
+                                    description={<div className="Description">
+                                    The official library to interact with Apify API from a web browser, Node.js, JavaScript, or Typescript applications.
+                                        <GitHubButton href="https://github.com/apify/apify-client-js" data-color-scheme="no-preference: light; light: light; dark: dark;" data-size="large" data-show-count="true">Star</GitHubButton>
```
Comment (jbartadev) [inline src/pages/api/index.tsx:170]: Seems like the GH button is not picking up the right background in light mode, works on the homepage


Snímek obrazovky 2024-05-05 v 12 28 37

File: src/pages/api/index.tsx [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+                    isCentered
+                    description={<>Apify API provides useful features like automatic retries and convenience functions.<br/>
+                It connects you with any Actors and their tasks either in the Store or your own.</>}
+                />
+                <SectionWrapper
+                    className={styles.LargerContent}
+                    heading="API reference"
+                    description={<div className="MainSectionContent">
+                    The Apify API allows developers to interact programmatically apps using HTTP requests.
+                    The Apify API is built around REST.
```
Comment (jbartadev) [inline src/pages/api/index.tsx:121]: Missing the links here - to REST

File: src/pages/api/index.tsx [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+                                `}/></ClientCodeWrapper>
+                                </SectionWrapper>
+                            ),
+                        },
+                    ]} />
+                </Section>
+                <Section heading="Related articles">
+                    <RelatedArticlesWrapper>
+                        <a href="https://blog.apify.com/web-scraping-with-client-side-vanilla-javascript/">
+                            <BlogArticle
```
Comment (jbartadev) [inline src/pages/api/index.tsx:247]: Wouldn't be better to define the IDs of the blog post and fetch them dynamically so the title and URL change if the blog post changes?

File: src/pages/api/index.tsx [hunk 6]
```diff
@@ -0,0 +1,266 @@
...
+# Fetch results from the actor run's default dataset
+dataset_items = apify_client.dataset(actor_call['defaultDatasetId']).list_items().items
+                                `}/></ClientCodeWrapper>
+                                </SectionWrapper>
+                            ),
+                        },
+                    ]} />
+                </Section>
+                <Section heading="Related articles">
+                    <RelatedArticlesWrapper>
```
Comment (jbartadev) [inline src/pages/api/index.tsx:245]: There should be 2 columns on the tablet based on the design

Snímek obrazovky 2024-05-05 v 12 35 41

**Review by jbartadev — COMMENTED** [review]
Looks good! A couple of notes.

Please bump the ui-library before merging as the current version has broken code highlighting In code block for dark mode

In the case of the gradient background - it's up to @hanatsai if she is fine with that. There is a way how to do it for sure

File: src/pages/api/index.tsx (hunk 5 above)
Comment (jbartadev) [reply src/pages/api/index.tsx:242]: Yeah, fair. Let's not complicate it

**Comment by hanatsai** [comment]
> Looks good! A couple of notes.
> 
> Please bump the ui-library before merging as the current version has broken code highlighting In code block for dark mode
> 
> In the case of the gradient background - it's up to @hanatsai if she is fine with that. There is a way how to do it for sure

I think it's fine, we can leave it as it is right now. :)

**Comment by hanatsai** [comment]
My fault 😭 the get started button should be green 😭😭😭 Sorry!! 
Screenshot 2024-05-06 at 17 12 03

File: nginx.conf [hunk 7]
```diff
@@ -190,7 +190,7 @@ server {
   rewrite ^/academy/web-scraping-for-beginners/crawling/processing-data$ /academy/web-scraping-for-beginners/crawling/exporting-data permanent;
   rewrite ^/academy/web-scraping-for-beginners/crawling/recap-collection-basics$ /academy/web-scraping-for-beginners/crawling/recap-extraction-basics permanent;
   rewrite ^/academy/php/using-apify-scraper-with-php$ /academy/php/use-apify-from-php permanent;
-  rewrite ^/api/?$ /api/v2 redirect;
```
Comment (HonzaTuron) [inline nginx.conf:193]: This is how we agreed with @B4nan how to redirect - we can rewrite original /v2. Also we want to redirect frome /api/client

**Comment by HonzaTuron** [comment]
@hanatsai Observations adressed.

**Review by hanatsai — APPROVED** [review]

File: src/pages/api/index.tsx (hunk 6 above)
Comment (jbartadev) [reply src/pages/api/index.tsx:254]: @HonzaTuron now it's 3 columns on the tablet (should be 2)

Snímek obrazovky 2024-05-07 v 13 20 29

File: src/pages/api/index.tsx [hunk 8]
```diff
@@ -0,0 +1,275 @@
...
+                        and uses standard HTTP response codes, authentication, and verbs.</p>
+                        <div>
+                            <Button to='/api/v2'>Check API reference</Button>
+                        </div>
+                    </div>}
+                >
+                    <ClientCodeWrapper>
+                        <CodeBlock content={[{ key: 'cURL', label: 'cURL', language: 'bash', code: `
+                # Prepare Actor input
+
```
Comment (jbartadev) [inline src/pages/api/index.tsx:141]: Let's remove the empty lines from the code examples (between the comment and code)

File: src/pages/api/index.tsx [hunk 9]
```diff
@@ -0,0 +1,275 @@
...
+                                    </div>}
+                                >
+                                    <ClientCodeWrapper>
+                                        <CodeBlock content='npm install apify-client' language="bash"/>
+                                        <CodeBlock
+                                            language='javascript'
+                                            hideLineNumbers
+                                            content={`
+// Easily run Actors, await them to finish using the convenient .call() method, and retrieve results from the resulting dataset.
+
```
Comment (jbartadev) [inline src/pages/api/index.tsx:193]: Also here, extra empty line between the comment and the other line of code

**Review by jbartadev — APPROVED** [review]
Preapproving, details to solve

**Review by baldasseva — CHANGES_REQUESTED** [review]
Missing link in the navbar
image

**Review by baldasseva — APPROVED** [review]
all good now from my side

File: nginx.conf [hunk 10]
```diff
@@ -190,7 +190,7 @@ server {
...
+  rewrite ^/api/client?$ /api redirect;
```
Comment (B4nan) [inline nginx.conf:193]: i think should be `/?`, meaning we redirect on `/api/client` and `/api/client/` but not `/api/client/xxx`. the way you have it means the `t` is optional, so its `/api/client` or `/api/clien`

**Pull request merged**